src/components/navigation-header/index.tsx

```diff
@@ -35,12 +34,13 @@ const RightSideHeaderContent = () => {
  */
 const NavigationHeader = () => {
   const { isDesktop } = useDeviceSize()
-  const currentPath = useCurrentPath({ excludeHash: true, excludeSearch: true })
-  const LeftSideHeaderContent =
-    currentPath === '/' ? HomePageHeaderContent : ProductPageHeaderContent
+  const currentProduct = useCurrentProduct()
+  const LeftSideHeaderContent = !currentProduct
+    ? HomePageHeaderContent
+    : ProductPageHeaderContent
 
   // TODO: menu for the home page, which does not use SidebarSidecarLayout
-  const shouldShowRightSide = !isDesktop && currentPath !== '/'
+  const shouldShowRightSide = !isDesktop && currentProduct
```

Comment:
I've modified the logic here to check if `currentProduct` data exists before trying to render elements that depend on that data.

Previously we made this decision purely on whether the URL was the home page - which meant for URLs not the home page, that don't have clear product context (eg `/yeehaw`), we'd have a client-side error when trying to render product-specific `RightSideHeaderContent`.